db/db_impl.cc

```diff
@@ -445,7 +445,7 @@ Status DBImpl::CloseHelper() {
   // CancelAllBackgroundWork called with false means we just set the shutdown
   // marker. After this we do a variant of the waiting and unschedule work
   // (to consider: moving all the waiting into CancelAllBackgroundWork(true))
-  CancelAllBackgroundWork(false);
+  CancelAllBackgroundWork(initial_db_options_.wait_for_background_work);
```

Comment:
Line 460 - 466 is supposed to do the work. If you see that the waiting missed any work, can you help debug how this can be triggered?

Reply:
@siying sorry, my mistake, it's not because of the `CloseHelper` didn't wait for ongoing background work. It seems that we destroy the column family before deleting the DB, compaction background was doing compaction and the column family was used in the compact filter.  

```c
Storage::~Storage() {
  db_->SyncWAL();
  for (auto handle : cf_handles_) db_->DestroyColumnFamilyHandle(handle);
  delete db_;
}
```

any way to avoid this problem? I would close the PR later.

Reply:
One workaround that doesn't destroy the column family, and allows some memory leak in the program exit. But it would be disturbed when running the memory leak detection, any pieces of advice?

thanks~ @siying

Reply:
I don't understand the question. I don't see any problem with the code you pasted. Users are expected to destroy column family handles before deleting the DB. What exact error did you see?

Reply:
Thanks for your reply, we occur the segment fault randomly when closing the DB, as the cfd of the column family was released.

```
Storage::~Storage() {
  db_->SyncWAL();
  for (auto handle : cf_handles_) db_->DestroyColumnFamilyHandle(handle);
  delete db_;
}
```
we destroy the column family firstly, and then delete the DB. But DB may do the compaction background work after destroying the column family, and compact filter was using  the column family, then cause segment fault when access the column family